src/app/shell/workshop-details/side-menu/actions/actions.component.ts

```diff
@@ -21,16 +22,18 @@ export class ActionsComponent implements OnInit {
   * to display button
   */
 
-  isDisplayed(): boolean {
+  isRegistered(): boolean {
     const user: User = this.store.selectSnapshot<User>(RegistrationState.user);
-    if (user && user.role === Role.provider) {
```

Comment:
get rid of isRegistered() function and instead use an isRegistered variable of the type boolean. On ngOnInit you can check the user role and then applies it to the isRegistered variable.